storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/integration/PutOperationTest.java

```diff
@@ -0,0 +1,340 @@
+
+        Assertions.assertThat(cursor).hasCount(1);
+
+        cursor.moveToFirst();
+
+        assertThat(testItem.equalsWithoutId(TestItem.fromCursor(cursor))).isTrue();
+
+        changesTestSubscriber.awaitTerminalEvent(60, SECONDS);
+        changesTestSubscriber.assertNoErrors();
+        changesTestSubscriber.assertValue(Changes.newInstance(TestItem.CONTENT_URI));
+    }
+
+    @Test
+    public void updateContentValuesExecuteAsBlocking() {
+        TestSubscriber<Changes> changesTestSubscriber = new TestSubscriber<Changes>();
+
+        storIOContentResolver
+                .observeChangesOfUri(TestItem.CONTENT_URI)
+                .take(2)
+                .subscribe(changesTestSubscriber);
+
+        Uri insertedUri = contentResolver.insert(TestItem.CONTENT_URI, TestItem.create(null, "value").toContentValues());
+
+        TestItem testItem = TestItem.create(ContentUris.parseId(insertedUri), "value2");
+
+        PutResult updateResult = storIOContentResolver
+                .put()
+                .contentValues(testItem.toContentValues())
+                .withPutResolver(testItemContentValuesPutResolver)
+                .prepare()
+                .executeAsBlocking();
+
+        assertThat(updateResult.wasUpdated()).isTrue();
+
+        Cursor cursor = contentResolver.query(TestItem.CONTENT_URI, null, null, null, null);
+
+        Assertions.assertThat(cursor).hasCount(1);
+
+        cursor.moveToFirst();
+
+        assertThat(testItem).isEqualTo(TestItem.fromCursor(cursor));
+
+        changesTestSubscriber.awaitTerminalEvent(60, SECONDS);
+        changesTestSubscriber.assertNoErrors();
+        changesTestSubscriber.assertValues(Changes.newInstance(TestItem.CONTENT_URI), Changes.newInstance(TestItem.CONTENT_URI));
+    }
+
+    @Test
+    public void updateContentValuesAsObservable() {
+        TestSubscriber<Changes> changesTestSubscriber = new TestSubscriber<Changes>();
+
+        storIOContentResolver
+                .observeChangesOfUri(TestItem.CONTENT_URI)
+                .take(2)
+                .subscribe(changesTestSubscriber);
+
+        Uri insertedUri = contentResolver.insert(TestItem.CONTENT_URI, TestItem.create(null, "value").toContentValues());
+
+        TestItem testItem = TestItem.create(ContentUris.parseId(insertedUri), "value2");
+
+        PutResult updateResult = storIOContentResolver
+                .put()
+                .contentValues(testItem.toContentValues())
+                .withPutResolver(testItemContentValuesPutResolver)
+                .prepare()
+                .createObservable()
+                .toBlocking()
+                .first();
+
+        assertThat(updateResult.wasUpdated()).isTrue();
+
+        Cursor cursor = contentResolver.query(TestItem.CONTENT_URI, null, null, null, null);
+
+        Assertions.assertThat(cursor).hasCount(1);
+
+        cursor.moveToFirst();
+
+        assertThat(testItem).isEqualTo(TestItem.fromCursor(cursor));
```

Comment:
Not required since Cursors closed in finalize :) I don't want to do it
manually, at least in tests :)

чт, 24 Сен 2015, 11:22, Dmitrii Nikitin <notifications@github.com>:

> In
> storio-content-resolver/src/test/java/com/pushtorefresh/storio/contentresolver/integration/PutOperationTest.java
> <https://github.com/pushtorefresh/storio/pull/472#discussion_r40295291>:
>
> > +                .contentValues(testItem.toContentValues())
> > +                .withPutResolver(testItemContentValuesPutResolver)
> > +                .prepare()
> > +                .createObservable()
> > +                .toBlocking()
> > +                .first();
> > +
> > +        assertThat(updateResult.wasUpdated()).isTrue();
> > +
> > +        Cursor cursor = contentResolver.query(TestItem.CONTENT_URI, null, null, null, null);
> > +
> > +        Assertions.assertThat(cursor).hasCount(1);
> > +
> > +        cursor.moveToFirst();
> > +
> > +        assertThat(testItem).isEqualTo(TestItem.fromCursor(cursor));
>
> Close cursor?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pushtorefresh/storio/pull/472/files#r40295291>.
>
-- 

@artem_zin